incubator/hnc/internal/reconcilers/object.go

```diff
@@ -650,20 +650,13 @@ func (r *ObjectReconciler) syncPropagation(ctx context.Context, log logr.Logger,
 // - Objects have a selector that doesn't match the destination namespace
 // - Service Account token secrets
 func (r *ObjectReconciler) shouldPropagateSource(log logr.Logger, inst *unstructured.Unstructured, dst string) bool {
-	selector, errSelector := selectors.GetSelector(inst)
-	// TODO: generate events for the errors
-	if errSelector != nil {
-		log.Error(errSelector, "Cannot propagate")
-	}
-	treeSelector, errTreeSelector := selectors.GetTreeSelector(inst)
-	if errTreeSelector != nil {
-		log.Error(errTreeSelector, "Cannot propagate")
-	}
-	noneSelector, errNoneSelector := selectors.GetNoneSelector(inst)
-	if errNoneSelector != nil {
-		log.Error(errNoneSelector, "Cannot propagate")
-	}
 	nsLabels := r.Forest.Get(dst).GetLabels()
+	isMatch, err := selectors.DoSelectorsMatch(inst, nsLabels)
+	if err != nil {
+		// TODO: generate events for the errors
+		log.Error(err, "Cannot propagate")
+		return false
+	}
```

Comment:
You could also be even more aggresive about early returns here:

```
if ok, err := selectors.ShouldPropagate(inst, nsLabels); err != nil {
  // ... what you have now ...
} else if !ok {
  return false
}
```

And never even make it to the switch statement.

Reply:
Done